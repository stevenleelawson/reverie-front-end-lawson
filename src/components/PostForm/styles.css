h1 {
  color: chartreuse;
}

form {
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 30vw;
}

input {
  margin: 1rem;
}
