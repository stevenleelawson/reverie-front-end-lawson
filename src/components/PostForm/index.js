import React, { Component } from 'react';
import './styles.css';

export default class PostForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      date_added: '',
      first_active: '',
      current_name: '',
      height: '',
      weight: '',
      intelligence_metric: 1
    }
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.props.addRobot(this.state)
    this.setState({
      date_added: '',
      first_active: '',
      current_name: '',
      height: '',
      weight: '',
      intelligence_metric: 1
    })
  }

  handleChange = (e) => {
    const { name, value} = e.target
    this.setState({
      [name]: value
    })
  }

  render() {
    const {
      date_added,
      first_active,
      current_name,
      height,
      weight,
      intelligence_metric
    } = this.state;

    return (
      <div className='form-container'>
        <h1>Add a new robot</h1>
        <form onSubmit={this.handleSubmit}>
          <input
            type='text'
            name='date_added'
            placeholder='date_added'
            value= { date_added }
            onChange={ this.handleChange }
          />
          <input
             type='text'
             name='first_active'
             value={ first_active }
             placeholder='first_active'
             onChange={ this.handleChange }
           />
          <input
             type='text'
             name='current_name'
             value={ current_name }
             placeholder='current_name'
             onChange={ this.handleChange }
           />
          <input
             type='text'
             name='height'
             placeholder='height'
             value={ height }
             onChange={ this.handleChange }
           />
          <input
             type='text'
             name='weight'
             placeholder='weight'
             value={ weight }
             onChange={ this.handleChange }
           />
          <input
             type='number'
             name='intelligence_metric'
             placeholder='intelligence_metric'
             value={ intelligence_metric }
             onChange={ this.handleChange }
           />
          <button className='add-robot' type='submit'>Add Robot</button>
        </form>
      </div>
    )
  }
}
