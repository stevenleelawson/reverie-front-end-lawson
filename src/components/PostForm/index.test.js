import React from 'react';
import { shallow } from 'enzyme';
import PostForm from './index';

describe('PostForm', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PostForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
});
