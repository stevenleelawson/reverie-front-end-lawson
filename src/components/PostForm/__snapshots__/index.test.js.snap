// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostForm should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Add a new robot
</h1>,
        <form
          onSubmit={[Function]}
>
          <input
                    name="date_added"
                    onChange={[Function]}
                    placeholder="date_added"
                    type="text"
                    value=""
          />
          <input
                    name="first_active"
                    onChange={[Function]}
                    placeholder="first_active"
                    type="text"
                    value=""
          />
          <input
                    name="current_name"
                    onChange={[Function]}
                    placeholder="current_name"
                    type="text"
                    value=""
          />
          <input
                    name="height"
                    onChange={[Function]}
                    placeholder="height"
                    type="text"
                    value=""
          />
          <input
                    name="weight"
                    onChange={[Function]}
                    placeholder="weight"
                    type="text"
                    value=""
          />
          <input
                    name="intelligence_metric"
                    onChange={[Function]}
                    placeholder="intelligence_metric"
                    type="number"
                    value={1}
          />
          <button
                    className="add-robot"
                    type="submit"
          >
                    Add Robot
          </button>
</form>,
      ],
      "className": "form-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add a new robot",
        },
        "ref": null,
        "rendered": "Add a new robot",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="date_added"
              onChange={[Function]}
              placeholder="date_added"
              type="text"
              value=""
/>,
            <input
              name="first_active"
              onChange={[Function]}
              placeholder="first_active"
              type="text"
              value=""
/>,
            <input
              name="current_name"
              onChange={[Function]}
              placeholder="current_name"
              type="text"
              value=""
/>,
            <input
              name="height"
              onChange={[Function]}
              placeholder="height"
              type="text"
              value=""
/>,
            <input
              name="weight"
              onChange={[Function]}
              placeholder="weight"
              type="text"
              value=""
/>,
            <input
              name="intelligence_metric"
              onChange={[Function]}
              placeholder="intelligence_metric"
              type="number"
              value={1}
/>,
            <button
              className="add-robot"
              type="submit"
>
              Add Robot
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "date_added",
              "onChange": [Function],
              "placeholder": "date_added",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "first_active",
              "onChange": [Function],
              "placeholder": "first_active",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "current_name",
              "onChange": [Function],
              "placeholder": "current_name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "height",
              "onChange": [Function],
              "placeholder": "height",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "weight",
              "onChange": [Function],
              "placeholder": "weight",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "intelligence_metric",
              "onChange": [Function],
              "placeholder": "intelligence_metric",
              "type": "number",
              "value": 1,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Robot",
              "className": "add-robot",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add Robot",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Add a new robot
</h1>,
          <form
            onSubmit={[Function]}
>
            <input
                        name="date_added"
                        onChange={[Function]}
                        placeholder="date_added"
                        type="text"
                        value=""
            />
            <input
                        name="first_active"
                        onChange={[Function]}
                        placeholder="first_active"
                        type="text"
                        value=""
            />
            <input
                        name="current_name"
                        onChange={[Function]}
                        placeholder="current_name"
                        type="text"
                        value=""
            />
            <input
                        name="height"
                        onChange={[Function]}
                        placeholder="height"
                        type="text"
                        value=""
            />
            <input
                        name="weight"
                        onChange={[Function]}
                        placeholder="weight"
                        type="text"
                        value=""
            />
            <input
                        name="intelligence_metric"
                        onChange={[Function]}
                        placeholder="intelligence_metric"
                        type="number"
                        value={1}
            />
            <button
                        className="add-robot"
                        type="submit"
            >
                        Add Robot
            </button>
</form>,
        ],
        "className": "form-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add a new robot",
          },
          "ref": null,
          "rendered": "Add a new robot",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="date_added"
                onChange={[Function]}
                placeholder="date_added"
                type="text"
                value=""
/>,
              <input
                name="first_active"
                onChange={[Function]}
                placeholder="first_active"
                type="text"
                value=""
/>,
              <input
                name="current_name"
                onChange={[Function]}
                placeholder="current_name"
                type="text"
                value=""
/>,
              <input
                name="height"
                onChange={[Function]}
                placeholder="height"
                type="text"
                value=""
/>,
              <input
                name="weight"
                onChange={[Function]}
                placeholder="weight"
                type="text"
                value=""
/>,
              <input
                name="intelligence_metric"
                onChange={[Function]}
                placeholder="intelligence_metric"
                type="number"
                value={1}
/>,
              <button
                className="add-robot"
                type="submit"
>
                Add Robot
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "date_added",
                "onChange": [Function],
                "placeholder": "date_added",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "first_active",
                "onChange": [Function],
                "placeholder": "first_active",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "current_name",
                "onChange": [Function],
                "placeholder": "current_name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "height",
                "onChange": [Function],
                "placeholder": "height",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "weight",
                "onChange": [Function],
                "placeholder": "weight",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "intelligence_metric",
                "onChange": [Function],
                "placeholder": "intelligence_metric",
                "type": "number",
                "value": 1,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Robot",
                "className": "add-robot",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Add Robot",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
